server/service/osquery_utils/queries.go

```diff
@@ -375,16 +371,16 @@ func ingestNetworkInterface(ctx context.Context, logger log.Logger, host *fleet.
 
 	// Attempt to extract public IP from the HTTP request.
 	ipStr := publicip.FromContext(ctx)
-	ip := net.ParseIP(ipStr)
-	if ip != nil {
-		if isPublicIP(ip) {
+	// First set host.PublicIP to empty to not hide an infrastructure change that
+	// misses to set or sets an invalid value in the expected HTTP headers.
+	host.PublicIP = ""
```

Comment:
Nit: Would be nice to rename this to just `IP`

Reply:
+1, probably worth doing when (if) we get complaints from customers that "Public IP address" in the UI is showing a private IP? (I don't know of a good name for this "IP"...)